### PR TITLE
RP2040 Demo

### PR DESCRIPTION
**Pull request by kilograham**
Adds demo applications (including standard main_full) for RP2040.

Note the same demo applications can be used with either the SMP or non SMP versions of the RP2040 kernel port (simply change the branch of the FreeRTOS/Source accordingly)

### CODE REVIEW

**Comment by aggarg** [comment]
Please move this project to `https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/ThirdParty`.

**Comment by kilograham** [comment]
> Please move this project to `https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/ThirdParty`.

where? just in that directory, in the new sub modules?

**Comment by aggarg** [comment]
> 
> 
> > Please move this project to `https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/ThirdParty`.
> 
> where? just in that directory, in the new sub modules?

Just in the directory. NOT in the submodules.

**Comment by aggarg** [comment]
My apologies - would you please move to https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/ThirdParty/Partner-Supported. 

Is it more involved than just moving files?

File: FreeRTOS/Demo/Common/Minimal/semtest.c [hunk 1]
```diff
@@ -231,7 +231,10 @@ short sError = pdFALSE, sCheckVariableToUse;
 			a cycle	(deliberately so to test the guarding) so will be starving
 			out lower priority tasks.  Block for some time to allow give lower
 			priority tasks some processor time. */
-			vTaskDelay( pxParameters->xBlockTime * semtstDELAY_FACTOR );
+			if (pxParameters->xBlockTime != (TickType_t) 0)
+			{
+				vTaskDelay(pxParameters->xBlockTime * semtstDELAY_FACTOR);
+			}
```
Comment (aggarg) [inline FreeRTOS/Demo/Common/Minimal/semtest.c:237]:
```suggestion
			if( pxParameters->xBlockTime != ( TickType_t ) 0 )
			{
				vTaskDelay( pxParameters->xBlockTime * semtstDELAY_FACTOR );
			}

```

Comment (kilograham) [reply FreeRTOS/Demo/Common/Minimal/semtest.c:237]: > A WIP PR was just merged. My apologies - would you please move to https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/ThirdParty/Partner-Supported.
> 
> Is it more involved than just moving files?

well i'm not sure anything here counts as "partner supported"

Comment (aggarg) [reply FreeRTOS/Demo/Common/Minimal/semtest.c:237]: Well, if you do not plan to address the issues reported in this demo, then you can move this to: https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/ThirdParty/Community-Supported

We are still in the process of publishing guidelines about where to put contributions. 

Thanks.

Comment (kilograham) [reply FreeRTOS/Demo/Common/Minimal/semtest.c:237]: I've moved it to Community-Supported for now

**Review by aggarg — DISMISSED** [review]

**Comment by kilograham** [comment]
do you want me to add "SMP" to the spell check list - will presumably require re-approval

**Comment by aggarg** [comment]
> 
> 
> do you want me to add "SMP" to the spell check list - will presumably require re-approval

Yes, please do.

**Review by aggarg — APPROVED** [review]

**Pull request merged**